### PR TITLE
Update dart-internal build names

### PR DESCRIPTION
**Pull request by drewroengoogle**
* Updates dart-internal build results to use the build properties name instead of the builder name. This is so that every task isn't called "Linux Production Engine Drone".

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: app_dart/lib/src/request_handlers/dart_internal_subscription.dart [hunk 1]
```diff
@@ -49,18 +49,30 @@ class DartInternalSubscription extends SubscriptionHandler {
...
 
     log.info(
       "Calling buildbucket api to get build data for build $buildbucketId",
     );
     final Build build = await _getBuildFromBuildbucket(request);
 
+    String? name;
+    if (build.input?.properties != null && build.input?.properties?["build"] != null) {
+      final Map<String, Object> buildProperties = build.input?.properties?["build"] as Map<String, Object>;
+      name = buildProperties["name"] as String;
```
Comment (godofredoc) [inline app_dart/lib/src/request_handlers/dart_internal_subscription.dart:71]: Is this property available in builds from both engine and flutter?

Comment (drewroengoogle) [reply app_dart/lib/src/request_handlers/dart_internal_subscription.dart:71]: Yes, [here's an example from the packaging_builder run on the drone.](https://ci.chromium.org/p/dart-internal/builders/flutter/Linux%20Production%20Engine%20Drone/19391?)

**Review by godofredoc — APPROVED** [review]

File: app_dart/lib/src/request_handlers/dart_internal_subscription.dart [hunk 2]
```diff
@@ -49,18 +49,30 @@ class DartInternalSubscription extends SubscriptionHandler {
...
+        fields:
+            "id,builder,number,createdBy,createTime,startTime,endTime,updateTime,status,input.properties,input.gitilesCommit",);
 
     log.info(
       "Calling buildbucket api to get build data for build $buildbucketId",
     );
     final Build build = await _getBuildFromBuildbucket(request);
 
+    String? name;
+    if (build.input?.properties != null && build.input?.properties?["build"] != null) {
```
Comment (godofredoc) [inline app_dart/lib/src/request_handlers/dart_internal_subscription.dart:69]: Please add a comment that this part applies only to engine_v2 builds

**Review by keyonghan — APPROVED** [review]
LGTM with nits.

**Pull request merged**